Description: Don't access free'd memory if a volume is unmounted whilst the dialog is displayed
Author: ?

diff -Nur -x '*.orig' -x '*~' gnome-settings-daemon-2.28.1/plugins/housekeeping/gsd-disk-space.c gnome-settings-daemon-2.28.1.new/plugins/housekeeping/gsd-disk-space.c
--- gnome-settings-daemon-2.28.1/plugins/housekeeping/gsd-disk-space.c	2009-08-12 10:15:00.000000000 +0100
+++ gnome-settings-daemon-2.28.1.new/plugins/housekeeping/gsd-disk-space.c	2009-11-04 00:09:42.505236169 +0000
@@ -181,7 +181,7 @@
         gint response;
         gboolean has_trash;
         gboolean retval = TRUE;
-        const gchar *path;
+        gchar *path;
 
         /* Don't show a dialog if one is already displayed */
         if (dialog)
@@ -190,7 +190,7 @@
         name = g_unix_mount_guess_name (mount->mount);
         free_space = (gint64) mount->buf.f_frsize * (gint64) mount->buf.f_bavail;
         has_trash = ldsm_mount_has_trash (mount);
-        path = g_unix_mount_get_mount_path (mount->mount);
+        path = g_strdup (g_unix_mount_get_mount_path (mount->mount));
 
         dialog = gsd_ldsm_dialog_new (other_usable_volumes,
                                       multiple_volumes,
@@ -214,7 +214,7 @@
                 break;
         case GSD_LDSM_DIALOG_RESPONSE_ANALYZE:
                 retval = FALSE;
-                ldsm_analyze_path (g_unix_mount_get_mount_path (mount->mount));
+                ldsm_analyze_path (path);
                 break;
         case GSD_LDSM_DIALOG_RESPONSE_EMPTY_TRASH:
                 retval = TRUE;
@@ -228,6 +228,8 @@
                 g_assert_not_reached ();
         }
 
+        g_free (path);
+
         return retval;
 }
 
